Mage/src/main/java/mage/choices/ChoiceImpl.java

```diff
@@ -311,7 +311,10 @@ public boolean setChoiceByAnswers(List<String> answers, boolean removeSelectAnsw
             // no key answer found, try to match by text starting with
             for (String needChoice : answers) {
                 for (Map.Entry<String, String> currentChoice : this.getKeyChoices().entrySet()) {
-                    if (currentChoice.getValue().startsWith(needChoice)) {
+                    String choiceValue = currentChoice.getValue();
+                    // Clean any html part (for easier unit test matching)
+                    String cleanedChoiceValue = choiceValue.replaceAll("<[^<>]*>", "");
+                    if (choiceValue.startsWith(needChoice) || cleanedChoiceValue.startsWith(needChoice)) {
```

Comment:
@JayDi85 fyi, I did alter the choice matching to strip the html part. Should not affect regular gameplay, but helps with making more precise tests on choosing a source by name.

Reply:
Look at changed tests. Any time I match `(source: Omniscience` (or other card name), that is what that kind of per-value choice, and with the Log name it is harder to properly match (has html for color + uuid).

Reply:
You can use starting text instead full ability:

![shot_240727_233730](https://github.com/user-attachments/assets/fe80df1d-fa1d-4413-81bb-dcf27aed828f)

Reply:
Yes, and I want to distinguish at times the text is the same but the source is different, hence the change. Like with the test with `Sakashima the Impostor` copying `Demon of Fate's Design`. It does matter.